robolectric-shadows/shadows-core/src/main/resources/org/robolectric/shadows/ShadowAccessibilityNodeInfo.java.vm

```diff
@@ -506,36 +834,54 @@ public class ShadowAccessibilityNodeInfo {
   }
 
   @Implementation
-  public void addAction (int action) {
-    switch (action) {
-      case AccessibilityNodeInfo.ACTION_CLICK:
-        setClickable(true);
-        break;
-      case AccessibilityNodeInfo.ACTION_LONG_CLICK:
-        setLongClickable(true);
-        break;
-      case AccessibilityNodeInfo.ACTION_SCROLL_FORWARD:
-        setScrollable(true);
-        break;
-#if ($api >= 18)
-      case AccessibilityNodeInfo.ACTION_SET_SELECTION:
-        setTextSelectionSetable(true);
-        break;
-      case AccessibilityNodeInfo.ACTION_PASTE:
-        setPasteable(true);
-        break;
+  public void addAction(int action) {
+#if ($api >= 21)
+  if ((action & (int)ReflectionHelpers.getStaticField(AccessibilityNodeInfo.class, "ACTION_TYPE_MASK")) != 0) {
```

Comment:
This reflection is brittle. Maybe wrap the reflection stuff in method calls and put them at the bottom of the file with comments to help whoever later updates to an Android version that break these calls. That person will likely be less irritated if s/he has some guidance on what's happening here.